File: leetcode3/변지협/v2/3471. Find the Largest Almost Missing Integer.py
'''
1. 아이디어 :
k만큼의 부분배열을 만들고, 부분배열에 포함되는 갯수 센다.

2. 시간복잡도 :
O(n*k)

3. 자료구조/알고리즘 :
'''
from collections import defaultdict
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        n = len(nums)
        dic = defaultdict(int)
        for i in range(n-k+1):
            sub = nums[i:i+k]
            st = set()
            for j in sub:
                st.add(j)
            for j in st:
                dic[j] += 1

        print(dic)
        
        try: 
            return max([key for key,value in dic.items() if value == 1])
        except:
            return -1

File: leetcode3/이진희/3471. Find the Largest Almost Missing Integer.java
/*

1. 아이디어 : 모든 k구간을 완전탐색으로 돌며 등장하는 구간 안 중복되지 않은 숫자를 각각 카운트, 가장 큰 수 반환 
              슬라이딩 윈도우 기법으로 풀려고 했지만, 중복된 숫자가 등장하는 경우를 처리하기가 까다로워 완전탐색으로 품

2. 시간복잡도 : O(N*K+50)

3. 자료구조/알고리즘 : 완전탐색, Set

 */

class Solution {
    public int largestInteger(int[] nums, int k) {
        int[] cnt = new int[51];
        for(int i=0; i<=nums.length-k; i++) {
            Set<Integer> set = new HashSet<>();
            for(int j=i; j<i+k; j++) {
                if(set.contains(nums[j])) continue;
                set.add(nums[j]);
                cnt[nums[j]]++;
            }
        }

        for(int i=50; i>=0; i--) {
            if(cnt[i] == 1) return i;
        }

        return -1;
    }
}